engine/core/src/test/java/org/datacleaner/test/full/scenarios/AnalyzerResultFutureAndAnalysisListenerTest.java

```diff
@@ -165,38 +165,57 @@ public void jobSuccess(AnalysisJob job, AnalysisJobMetrics metrics) {
-
-                        // the .await() method should not return before
-                        // jobSuccess()
-                        + "jobSuccess, AnalysisResultFuture.await() returned, "
-
-                        // the analyzer result future should be ready at this
-                        // point
-                        + "AnalyzerResultFuture.isReady() (2) = true]", allMessages);
+        // export to DS, dup id in RECORDS
+
+        final String originalMessagesString = messagesList.toString();
+
+        final int indexRunReturned = getIndexAndVerifyExists(messagesList, "AnalysisRunner.run(job) returned");
+        final int indexRowProcessingSuccess = getIndexAndVerifyExists(messagesList, "rowProcessingSuccess");
+
+        assertTrue(originalMessagesString, indexRowProcessingSuccess > indexRunReturned);
+
+        final int indexNormalAnalyzerSuccess = getIndexAndVerifyExists(messagesList,
+                "componentSuccess(normal analyzer,ListResult)");
+        final int indexFutureAnalyzerSuccess = getIndexAndVerifyExists(messagesList,
+                "componentSuccess(analyzer with future result,AnalyzerResultFutureImpl)");
+
+        assertTrue(originalMessagesString, indexNormalAnalyzerSuccess > indexRowProcessingSuccess);
+        assertTrue(originalMessagesString, indexFutureAnalyzerSuccess > indexRowProcessingSuccess);
+
+        final int indexAnalyzerResultNotReady = getIndexAndVerifyExists(messagesList,
+                "AnalyzerResultFuture.isReady() (1) = false");
+        final int indexListener1Success = getIndexAndVerifyExists(messagesList,
+                "AnalyzerResultFuture.Listener (1).onSuccess()");
+        final int indexListener2Success = getIndexAndVerifyExists(messagesList,
+                "AnalyzerResultFuture.Listener (2).onSuccess()");
+
+        assertTrue(originalMessagesString, indexListener1Success > indexAnalyzerResultNotReady);
+        assertTrue(originalMessagesString, indexListener2Success > indexAnalyzerResultNotReady);
+
+        final int indexAnalysisResultFutureAwaitReturned = getIndexAndVerifyExists(messagesList,
+                "AnalysisResultFuture.await() returned");
+
+        assertTrue(originalMessagesString, indexAnalysisResultFutureAwaitReturned > indexListener1Success);
+        assertTrue(originalMessagesString, indexAnalysisResultFutureAwaitReturned > indexListener2Success);
+
+        final int indexAnalyzerResultReady = getIndexAndVerifyExists(messagesList,
+                "AnalyzerResultFuture.isReady() (2) = true");
+
+        assertTrue(originalMessagesString, indexAnalyzerResultReady > indexAnalyzerResultNotReady);
+
+        final int indexJobSuccess = getIndexAndVerifyExists(messagesList, "jobSuccess");
+
+        assertTrue(originalMessagesString, indexJobSuccess > indexFutureAnalyzerSuccess);
```

Comment:
I know the JUnit message parameter, it's just a using it for something else than an actual message like "The list should have had stuff in it" :)